src/app/components/dynamic-hub/dynamic-hub.component.ts

```diff
@@ -366,6 +387,34 @@ export class DynamicHubComponent implements OnInit {
         } else {
             this.cards = this.cardsRef;
         }
+
+        this.dashboardSearch.buildSearchIndex(this.cards);
+        this.filterDashboard(this.dashboardSearch.currentFilter);
+    }
+
+    public filterDashboard(searchValue: string, preserveEmptyCardList = false){
+        let previous;
+        if(searchValue){
+            previous = this.cardsFiltered;
+            this.cardsFiltered = new Map<string, Card[]>();
+            this.dashboardSearch
+                .getMatchingSearchIndexes()
+                .forEach(searchIndex => {
+                    if(this.cardsFiltered.has(searchIndex.key)){
+                        this.cardsFiltered.get(searchIndex.key).push(this.cards.get(searchIndex.key)[searchIndex.cardIndex]);
```

Comment:
could you add a check if **this.cards.get(key)** exists ?
It is possible that the `this.cards` map gets reset by a refresh call and be for an instant empty

Reply:
a refresh triggered by a button ?